### PR TITLE
move name and id out of Feature.init

### PR DESCRIPTION
**Pull request by cpoerschke**
(towards fully pluggable Feature classes)

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java [hunk 1]
```diff
@@ -141,6 +134,16 @@ public int getId() {
     return id;
   }
 
+  /**
+   * @param id
+   *          Unique ID for this feature. Similar to feature name, except it can
+   *          be used to directly access the feature in the global list of
+   *          features.
+   */
+  public void setId(int id) {
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java]: What's the reason for ripping the id and not keeping it together with the name in the constructor?

**Comment by mnilsson23** [comment]
new develop branch was pushed due to solr conflicts, please reopen the request against the develop branch again after rebasing on top of it

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java (hunk 1 above)
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java]: The name is an actual part of the feature (and its json) but the id is indirect i.e. calculated as the index of the feature in the store. So I kept id out of the constructor signature, at least for now, with a view towards seeing if later perhaps we can get rid of Feature.id somehow.

**Comment by cpoerschke** [comment]
Just to be clear, is `master-ltr-plugin-develop` still the name of the develop branch i.e. it's a new branch but with the old name, or will there be a new branch with a new name?

**Comment by mnilsson23** [comment]
You are correct, `master-ltr-plugin-develop` is the branch to use, which now has the newest updates pushed to it.

**Comment by cpoerschke** [comment]
https://github.com/bloomberg/lucene-solr/pull/50 is the requested replacement for this pull request here.

**Pull request closed without merging**